src/app/modules/data-set/instances/instances.component.ts

```diff
@@ -0,0 +1,320 @@
+import { FormControl, Validators } from "@angular/forms";
+import { MatDialog } from "@angular/material/dialog";
+import { MatTableDataSource } from "@angular/material/table";
+import { Router } from "@angular/router";
+import { DialogConfigService } from "../dialogs/dialog-config.service";
+import { DisagreeingAnnotationsDialogComponent } from "../dialogs/disagreeing-annotations-dialog/disagreeing-annotations-dialog.component";
+import { Annotation } from "../model/annotation/annotation.model";
+import { DataSetProject } from "../model/data-set-project/data-set-project.model";
+import { DataSet } from "../model/data-set/data-set.model";
+import { AnnotationStatus } from "../model/enums/enums.model";
+import { Instance } from "../model/instance/instance.model";
+import { AnnotationService } from "../services/annotation.service";
+import { DataSetProjectService } from "../services/data-set-project.service";
+import { NotificationService } from "../services/shared/notification.service";
+import { LocalStorageService } from "../services/shared/local-storage.service";
+import { Subscription } from "rxjs";
+
+@Component({
+    selector: 'de-instances',
+    templateUrl: './instances.component.html',
+    styleUrls: ['./instances.component.css']
+})
+  
+export class InstancesComponent implements OnInit {
+
+    public chosenProject: DataSetProject = new DataSetProject();
+    public chosenDataset: DataSet = new DataSet();
+    public chosenInstance: Instance = new Instance(this.storageService);
+
+    public dataSource: MatTableDataSource<Instance> = new MatTableDataSource<Instance>();
+    private initColumns: string[] = ['codeSnippetId', 'annotated', 'severity'];
+    public displayedColumns: string[] = this.initColumns;
+    public searchInput: string = '';
+    public selectedAnnotationStatus: AnnotationStatus = AnnotationStatus.All;
+    public annotationStatuses: string[] = Object.keys(AnnotationStatus);
+    public selectedSeverity: number | null = null;
+    public severityValues: Set<number|null> = new Set();
+    public codeSmells: string[] = [];
+    public selectedSmellFormControl = new FormControl('', Validators.required);
+    public filter: string = 'All instances';
+    
+    private datasetChosenSub: Subscription | undefined;
+    private projectChosenSub: Subscription | undefined;
+    private instanceChosenSub: Subscription | undefined;
+    private nextInstanceSub: Subscription | undefined;
+    private previousInstanceSub: Subscription | undefined;
+    private newAnnotationSub: Subscription | undefined;
+    private changedAnnotationSub: Subscription | undefined;
+    
+    constructor(private dialog: MatDialog, private router: Router,
+      public storageService: LocalStorageService, private annotationService: AnnotationService,
+      private annotationNotificationService: NotificationService,
+      private projectService: DataSetProjectService, private location: Location) {
+    }
+
+    ngOnInit(): void {
```

Comment:
This component seems very complex. Can you come up with a way to simplify it?

Would the code become simpler if we group the subscriptions into a single subscription that can process multiple events (and act based on the event type)?